### ISSUE TITLE
Update engine script create_ios_framework to optionally take ARM simulator path

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#73066]
In preparation for [an ARM simulator framework](https://github.com/flutter/flutter/issues/64502) update [create_ios_framework](https://github.com/flutter/engine/blob/06f9654c67f80711a8c7970ede61c328bdf999a6/sky/tools/create_ios_framework.py#L23) to replace `--simulator-out-dir` with `--simulator-x86-out-dir` and `--simulator-arm64-out-dir`.  `lipo` the simulator frameworks together.

When that's done, update the [recipe logic that calls this script](https://flutter.googlesource.com/recipes/+/refs/heads/master/recipes/engine.py#1104) to pass in `--simulator-x86-out-dir`. 
 When https://github.com/flutter/flutter/issues/64502 is ready, it will additionally pass in `--simulator-arm64-out-dir`.

Don't remove deprecated `--simulator-out-dir` until after the recipe is updated to avoid breaking changes.

**Comment by gw280** [comment]
Quick update: the engine scripts are ready and mostly work (I just need to do a minor fix and re-land https://github.com/flutter/buildroot/pull/437), but there's an issue that needs to be resolved on CI before we can modify the recipes.

In order to build for arm64 simulator, we need to use Xcode 12.0.0 or newer, and thus we can't use our own llvm. As a result, we can't use goma to build, which means the build takes longer. Adding a new arm64 simulator build step to the existing iOS builder hits the timeout ceiling and so the build fails.

After speaking with @godofredoc and @CaseyHillers, it looks like they're working on refactoring the existing CI recipes to allow for these builds to be sharded out and have cross-dependencies, which we would need in order to add the arm64 simulator builds.

**Comment by gw280** [comment]
Closing in favour of https://github.com/flutter/flutter/issues/84672 as the other work is all done/merged.
